adagrams/game.py

```diff
@@ -1,11 +1,124 @@
+from random import choice
+
+
 def draw_letters():
-    pass
+
+    MAX_LETTERS_IN_HAND = 10
+
+    LETTER_POOL = {
+        'A': 9,
+        'B': 2,
+        'C': 2,
+        'D': 4,
+        'E': 12,
+        'F': 2,
+        'G': 3,
+        'H': 2,
+        'I': 9,
+        'J': 1,
+        'K': 1,
+        'L': 4,
+        'M': 2,
+        'N': 6,
+        'O': 8,
+        'P': 2,
+        'Q': 1,
+        'R': 6,
+        'S': 4,
+        'T': 6,
+        'U': 4,
+        'V': 2,
+        'W': 2,
+        'X': 1,
+        'Y': 2,
+        'Z': 1
+    }
+    
+    letter_pool_list = []
+
+    # Creating a letter pool list to emulate a pile of letters to choose from:
+    for letter in LETTER_POOL:
+         temp = [letter] * LETTER_POOL[letter]
+         letter_pool_list.extend(temp)
+
+    #build the player's hand (letters)
+    letters = []
+    while len(letters) < MAX_LETTERS_IN_HAND:
+        # choose a random letter from a list of LETTER_POOL keys using random.choice
+        rand_letter = choice(letter_pool_list)
+        if letters.count(rand_letter) >= LETTER_POOL[rand_letter]:
+            continue
+        else:
+            letters.append(rand_letter)
+
+    return letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()  # forces the input to be case insensitive
+
+    # to ensure each letter is in letter_bank and does not occur too frequently:
+    for letter in word:
```

Comment:
Nice job here as well. This is another scenario in which mutation is avoided at the expense of a complex condition. I think it's simpler to create a shallow copy of `letter_bank`, then check whether the letter is in the word, removing it from the letter bank copy if so (if not, we've encountered a letter not in the letter bank, so we can return false).
```python
    letter_bank_copy = letter_bank[:]
    for letter in word:
        if letter in letter_bank_copy:
            letter_bank_copy.remove(letter)
        else:
            return False

    return True
```